Hide update/version info when auto-update disabled

michalkleiner opened the pull request:
### Description:

Ref. DEV-17254.

Disabling the update info widget completely when auto updates are disabled.

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

sgiehl reviewed (dismissed):
As mentioned before, this might not be the approach we want to go with. It might fix the current problem, but has the potential to cause other problems in the future.
There could e.g. be an element where the child element isn't displayed on purpose until some data was loaded or similar...

michalkleiner committed:
Hide the update info in top controls if auto update disabled

michalkleiner commented:
Updated approach as discussed to disable the whole update info widget when auto update is disabled.

michalkleiner committed:
Let auto-update enabled control the whole logic

The isAutoUpdateEnabled flag actually means the instance can talk to Matomo servers at all, so there wouldn't be any checks
if it was false anyway.

mneudert commented on plugins/CoreHome/templates/_headerMessage.twig:
```diff
@@ -8,12 +8,14 @@
-        and (
-            (isManualUpdateCheck is defined and isManualUpdateCheck and lastUpdateCheckFailed is defined and lastUpdateCheckFailed)
-            or (isAdminArea is defined and isAdminArea)
+    isAutoUpdateEnabled and (
```
Should we also open a PR for Cloud to remove the no longer necessary `display: none` on their side? Or just ping them?

michalkleiner replied:
I thought about it, we can just ping them.

michalkleiner replied:
ref. PG-3097
